alibi_detect/od/sr.py

```diff
@@ -5,15 +5,31 @@
 
 logger = logging.getLogger(__name__)
 
+# numerical stability for division
 EPSILON = 1e-8
 
+# padding values
+PADDING_CONSTANT = 'constant'
+PADDING_REPLICATE = 'replicate'
+PADDING_REFLECT = 'reflect'
+PADDINGS = [PADDING_CONSTANT, PADDING_REFLECT, PADDING_REPLICATE]
+
+# padding sides
+SIDE_BILATERAL = 'bilateral'
+SIDE_LEFT = 'left'
+SIDE_RIGHT = 'right'
+SIDES = [SIDE_BILATERAL, SIDE_RIGHT, SIDE_LEFT]
+
 
 class SpectralResidual(BaseDetector, ThresholdMixin):
 
     def __init__(self,
                  threshold: float = None,
                  window_amp: int = None,
                  window_local: int = None,
+                 padding_amp_method: str = PADDING_REFLECT,
+                 padding_local_method: str = PADDING_REFLECT,
+                 padding_amp_side: str = SIDE_BILATERAL,
```

Comment:
Since this is user facing, I think we should be more explicit, e.g. types as `Literal['constant', 'replicate', 'reflect'] = 'reflect'` which unfortunately introduces some name duplication (unless there's a way to define `Literal` types given a list of string constant from the padding enum?) but is likely worth it for clarity.